stan/math/prim/scal/prob/beta_binomial_lcdf.hpp

```diff
@@ -103,10 +103,9 @@ namespace stan {
         const T_partials_return mu = alpha_dbl + n_dbl + 1;
         const T_partials_return nu = beta_dbl + N_dbl - n_dbl - 1;
 
-        const T_partials_return F = F32((T_partials_return)1, mu,
-                                                    -N_dbl + n_dbl + 1,
-                                                    n_dbl + 2, 1 - nu,
-                                                    (T_partials_return)1);
+        T_partials_return F;
+        F = F32((T_partials_return)1, mu, -N_dbl + n_dbl + 1, n_dbl + 2,
+          1 - nu, (T_partials_return)1);
```

Comment:
OK.  I'm still not sure how these functions are getting instantiated.  What you have isn't wrong, it just uses C-style casts and calls constructors more than once.

For future reference, C++ broke C-style casts down into dynamic and static variants.  You're going to be using static casting here, so the C-style `(foo)a` will do the same thing as  the C++ style `static_cast<foo>(a)`.  If `a` is an integer, then the only way that cast will resolve is as a constructor, so that you can also write `foo(a)` here.  Then once you see that, you can see that you can save the version and pass.  It can be dangerous to cast in an argument if it's being passed by reference because the lifespan of the constructed `foo` will only be the function call.

writing  will do the same thing as `foo(a)` in most cases, because the cast will be resolved as a constructor.  You can also write `static_cast<foo>(a)` if you prefer the cast-like style.

 What happens with a